File: semana-04/soluciones/Alexey-Examen parcial/uq-ai-frontend/src/types/lead.ts
export interface Lead {
    id: number;
    nombre: string;
    email: string;
    empresa?: string;
    telefono?: string;
    mensaje?: string;
    fechaRegistro: string;
}

export interface LeadRequest {
    nombre: string;
    email: string;
    empresa: string;
    telefono: string;
    mensaje: string;
}

File: semana-04/soluciones/Alexey-Examen parcial/backend/src/main/java/com/uqai/entity/Lead.java
package com.uqai.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "leads")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Lead {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String nombre;

    @Email
    private String email;

    private String empresa;

    private String telefono;

    @Column(length = 1000)
    private String mensaje;

    private LocalDateTime fechaRegistro;
}